src/Shared/Buffers.MemoryPool/DefaultMemoryPoolFactory.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Connections;
+
+namespace System.Buffers;
+
+#nullable enable
+
+internal sealed class DefaultMemoryPoolFactory : IMemoryPoolFactory<byte>, IDisposable
+{
+    private readonly IMeterFactory _meterFactory;
+    private readonly ConcurrentDictionary<PinnedBlockMemoryPool, PinnedBlockMemoryPool> _pools = new();
+    private readonly PeriodicTimer _timer;
+
+    public DefaultMemoryPoolFactory(IMeterFactory? meterFactory = null)
+    {
+        _meterFactory = meterFactory ?? NoopMeterFactory.Instance;
+        _timer = new PeriodicTimer(TimeSpan.FromSeconds(10));
+        _ = Task.Run(async () =>
+        {
+            try
+            {
+                while (await _timer.WaitForNextTickAsync())
+                {
+                    foreach (var pool in _pools.Keys)
+                    {
+                        pool.PerformEviction();
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex);
+            }
+        });
+    }
+
+    public MemoryPool<byte> Create()
+    {
+        var pool = new PinnedBlockMemoryPool(_meterFactory);
+
+        _pools.TryAdd(pool, pool);
+
+        pool.DisposeCallback = (self) =>
```

Comment:
Maybe not a big deal, but we can avoid the closure if you passed in state to the registration 😄